### ISSUE TITLE
`com.intellij.diagnostic.PluginException: 2102 ms to call on EDT ResetSelectedToRemoteAction#update@GoToAction`

### ISSUE THREAD

**Issue by shakil-jc** [VirtusLab/git-machete-intellij-plugin#1692]
## Running environment
- Git Machete plugin version - 4.0.0
- IDE - PhpStorm 2023.2
- Operating system - Windows 10 10.0

## Bug description
**Please include steps to reproduce (like `go to...`/`click on...` etc.) + expected and actual behaviour**.

## (optional) Screenshots
If applicable, add screenshots (or screen recordings, see [Peek](https://github.com/phw/peek#peek---an-animated-gif-recorder) on Linux)
to help explain your problem.

## IDEA - additional info
N/A

## IDEA - stack trace
```
com.intellij.diagnostic.PluginException: 2102 ms to call on EDT ResetSelectedToRemoteAction#update@GoToAction (com.virtuslab.gitmachete.frontend.actions.contextmenu.ResetSelectedToRemoteAction). Revise AnAction.getActionUpdateThread property [Plugin: com.virtuslab.git-machete]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)...
	Suppressed: java.lang.Throwable: EDT-trace-at-697-ms (5/5)
		at java.base@17.0.7/java.util.zip.Inflater.inflateBufferBuffer(Native Method)...
	Suppressed: java.lang.Throwable: EDT-trace-at-633-ms (4/5)
		at java.base@17.0.7/java.util.zip.Inflater.inflateBufferBuffer(Native Method)...
	Suppressed: java.lang.Throwable: EDT-trace-at-528-ms (3/5)
		at java.base@17.0.7/java.util.zip.Inflater.inflateBufferBuffer(Native Method)...
	Suppressed: java.lang.Throwable: EDT-trace-at-238-ms (2/5)
		at java.base@17.0.7/java.lang.ClassLoader.defineClass2(Native Method)...
	Suppressed: java.lang.Throwable: EDT-trace-at-112-ms (1/5)
		at java.base@17.0.7/java.util.zip.Inflater.inflateBufferBuffer(Native Method)
		at java.base@17.0.7/java.util.zip.Inflater.inflate(Inflater.java:546)
		at app//com.intellij.util.lang.ImmutableZipEntry.getByteBuffer(ImmutableZipEntry.java:140)
		at app//com.intellij.util.lang.HashMapZipFile.getByteBuffer(HashMapZipFile.java:100)
		at app//com.intellij.util.lang.ZipResourceFile.findClass(ZipResourceFile.java:107)
		at app//com.intellij.util.lang.JarLoader.findClass(JarLoader.java:58)
		at app//com.intellij.util.lang.ClassPath.findClassInLoader(ClassPath.java:238)
		at app//com.intellij.util.lang.ClassPath.findClass(ClassPath.java:188)
		at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.kt:326)
		at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.kt:178)
		at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.kt:151)
		at java.base@17.0.7/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
		at com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.getOngoingSideEffectingActions(BaseProjectDependentAction.java:43)
		at com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.update(BaseProjectDependentAction.java:66)
		at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performDumbAwareUpdate$0(ActionUtil.java:159)
		at com.intellij.openapi.actionSystem.ex.ActionUtil$$Lambda$1124/0x0000000100dba0d0.run(Unknown Source)
		at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:182)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.doUpdate(ActionUpdater.java:740)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$updateActionReal$4(ActionUpdater.java:145)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$1242/0x0000000100ef28d8.get(Unknown Source)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$7(ActionUpdater.java:240)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$2650/0x00000001018d5108.compute(Unknown Source)
		at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:71)
		at com.intellij.openapi.progress.ProgressManager$$Lambda$1723/0x000000010131a880.run(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$1234/0x0000000100ef6ea8.run(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
		at com.intellij.openapi.progress.impl.CoreProgressManager$$Lambda$1235/0x0000000100ef7310.compute(Unknown Source)
		at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
		at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
		at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
		at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
		at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
		at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:71)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$8(ActionUpdater.java:231)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$2649/0x00000001018d4ec0.invoke(Unknown Source)
		at com.intellij.platform.diagnostic.telemetry.helpers.TraceKt.computeWithSpan(trace.kt:171)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater.lambda$computeOnEdt$9(ActionUpdater.java:229)
		at com.intellij.openapi.actionSystem.impl.ActionUpdater$$Lambda$2647/0x00000001018d4a30.get(Unknown Source)
		at io.opentelemetry.context.Context.lambda$wrapSupplier$8(Context.java:309)
		at io.opentelemetry.context.Context$$Lambda$2648/0x00000001018d4c78.get(Unknown Source)
		at com.intellij.openapi.actionSystem.impl.ActionUpdateEdtExecutor.lambda$computeOnEdt$0(ActionUpdateEdtExecutor.java:47)
		at com.intellij.openapi.actionSystem.impl.ActionUpdateEdtExecutor$$Lambda$1256/0x0000000100f03038.run(Unknown Source)
		at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:193)
		at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:861)
		at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:478)
		at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:79)
		at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:121)
		at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:41)
		at com.intellij.openapi.application.impl.FlushQueue$$Lambda$555/0x000000010061ac50.run(Unknown Source)

**Comment by PawelLipski** [comment]
Huh looks that UI froze on class loading... while the performance of class loading is obviously beyond our control, we could just move action updates out of UI thread (via `AnAction.getActionUpdateThread`, as suggested). This will, however, complicate the action code a bit... so let's refrain for a moment. If the problem re-occurs for you @shakil-jc, or there are more reports like that coming in, we'll think of introducing the changes.

**Comment by PawelLipski** [comment]
Opened https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1695, to be released soon as v4.0.1
